### PR TITLE
feat(reviews): conditionally enable reviews functionality based on store setting

### PR DESCRIPTION
**Pull request by matthewvolk**
## What/Why?

This PR conditionally enables storefront reviews functionality based on the `site.settings.reviews.enabled` GraphQL setting. The behavior matches Stencil + Cornerstone: when reviews are disabled, rating displays on PDP and PLP, submissions on PDP, and the reviews section on PDP are hidden.

- Added `reviews.enabled` queries to product, search, category, and brand page data fetchers
- Introduced `reviewsEnabled` prop throughout the product display component hierarchy:
  - `ProductCard` - conditionally renders rating display
  - `ProductDetail` - conditionally renders rating display
  - `ProductList` and `ProductsListSection` - pass `reviewsEnabled` to child `ProductCard` components
- Product detail page conditionally renders the `Reviews` component based on the setting
- All pages that display products (search, category, brand, product detail) fetch and propagate the `reviewsEnabled` flag

The implementation defaults to `false` when the setting is unavailable, ensuring backward compatibility and graceful degradation.

## Testing

1. **With Reviews Enabled:**
   - Navigate to a product detail page - verify rating displays in product header and reviews section is visible
   - Navigate to search/category/brand pages - verify product cards show ratings when available
   - Verify ratings only display when `rating > 0`

2. **With Reviews Disabled:**
   - Disable reviews in BigCommerce admin settings
   - Navigate to product detail page - verify no rating in header and no reviews section
   - Navigate to search/category/brand pages - verify product cards do not show ratings
   - Verify no GraphQL errors occur

3. **Edge Cases:**
   - Test with products that have no reviews (rating = 0 or null) - should not display rating even when reviews are enabled
   - Test with products that have reviews - should display rating when reviews are enabled

When reviews are disabled:
reviews-disabled-plp
reviews-disabled-pdp-summary
reviews-disabled-pdp-details

When reviews are enabled:
reviews-enabled-plp
reviews-enabled-pdp-details
reviews-enabled-pdp-details

## Migration

**For developers with customized Catalyst forks:**

1. **GraphQL Queries:** If you've customized any of the following page data files, you'll need to add the `reviews.enabled` query:
   - `core/app/[locale]/(default)/product/[slug]/page-data.ts` - Add `reviews { enabled }` to `ProductQuery`
   - `core/app/[locale]/(default)/(faceted)/search/page-data.ts` - Already includes the query
   - `core/app/[locale]/(default)/(faceted)/category/[slug]/page-data.ts` - Already includes the query
   - `core/app/[locale]/(default)/(faceted)/brand/[slug]/page-data.ts` - Already includes the query

2. **Component Props:** If you've customized any of these components, you may need to add the `reviewsEnabled` prop:
   - `core/vibes/soul/primitives/product-card/index.tsx` - Optional prop, defaults to `false`
   - `core/vibes/soul/sections/product-detail/index.tsx` - Optional prop on product object
   - `core/vibes/soul/sections/product-list/index.tsx` - Optional prop, defaults to `undefined`
   - `core/vibes/soul/sections/products-list-section/index.tsx` - Optional prop, defaults to `undefined`

3. **Conditional Rendering:** If you've customized the product detail page (`core/app/[locale]/(default)/product/[slug]/page.tsx`), ensure the `Reviews` component is conditionally rendered:
   ```tsx
   {reviewsEnabled && (
     <Reviews ... />
   )}
   ```

4. **No Breaking Changes:** All `reviewsEnabled` props are optional with safe defaults, so existing customizations will continue to work (reviews will simply not display until the prop is added).

---

Note: This pull request description was generated with the assistance of AI.

### CODE REVIEW

File: core/vibes/soul/primitives/product-card/index.tsx [hunk 1]
```diff
@@ -149,6 +153,9 @@ export function ProductCard({
               </span>
             )}
             {price != null && <PriceLabel colorScheme={colorScheme} price={price} />}
+            {reviewsEnabled && typeof rating === 'number' && rating > 0 && (
```
Comment (matthewvolk) [inline core/vibes/soul/primitives/product-card/index.tsx:156]: Elected to not render the `Rating` component if a product has no reviews. Otherwise, all product cards would contain 5 empty stars, which visually looks like they're all rated as 0.

File: core/vibes/soul/primitives/product-card/index.tsx [hunk 2]
```diff
@@ -55,7 +58,8 @@ export interface ProductCardProps {
  * ```
  */
 export function ProductCard({
-  product: { id, title, subtitle, badge, price, image, href, inventoryMessage },
+  product: { id, title, subtitle, badge, price, image, href, inventoryMessage, rating },
+  reviewsEnabled = false,
```
Comment (matthewvolk) [inline core/vibes/soul/primitives/product-card/index.tsx:62]: Is `false` the right default here? I used `false` simply to match the default value for `showCompare` below. I guess I'm not entirely sure of all the scenarios in which GQL returns `null` for the settings field (that's the only reason that this prop would receive `undefined`)... and therefore am not sure if `false` is the best default.

Comment (bookernath) [reply core/vibes/soul/primitives/product-card/index.tsx:62]: It's a tradeoff. `false` protects against accidental over-exposure of data in unexpected scenarios or failure scenarios, which seems like a better option than a fail-open. In practice we expect a value to always be returned.

File: core/vibes/soul/primitives/product-card/index.tsx (hunk 1 above)
Comment (bookernath) [reply core/vibes/soul/primitives/product-card/index.tsx:156]: nice touch
